### ISSUE TITLE
Hide focus outline

### ISSUE THREAD

**Issue by steventilator** [grommet/grommet#3642]
Sometimes the focus outline stays visible after clicking on a link/button/anchor (see screenshot). This happens when navigating by mouse and touch, not keyboard. It happens on Chrome but not on Safari and Firefox.

Is there a way to hide the outline?

<img width="258" alt="focus-outline" src="https://user-images.githubusercontent.com/39826675/71725074-f0b72b80-2e32-11ea-9bcf-ff24459c5220.png">

- Grommet version: 2.9.0
- Browser Name and version: Chrome 79.0.3945.88 
- Operating System and version (desktop or mobile): Various

**Comment by steventilator** [comment]
Okay, I realized that I can set `focusIndicator={false}` to hide it. Is there a way to set this on every Button or Box? Or do I have to add it manually to every instance?

**Comment by CR1AT0RS** [comment]
I agree focus is also not consistent across Mobile UI so I think it should be optional.

**Comment by iMerica** [comment]
TLDR from Slack Discussion:

`focusIndicator` needs to be false by default and it needs to update its definition as "focus indicator that displays after clicking something". And add a separate capability for displaying a focus indicator when tabbing through the page.

**Comment by srallen** [comment]
For accessibility, it's actually best to typically have the styles for the click, hover, and focus events to be the same both for maintainability and predictability for users who use a variety of assistive tools. Disabling a focus indicator and or separating the styles between keyboard and mouse events inadvertently create less accessible components. Any configurability supported here should be done with great caution. 

I would agree that there's a bug with the focus outline persisting after the click event, but I don't think the solution is to necessarily allow devs to configure this to potentially make less accessible components.

**Comment by srallen** [comment]
Caution should be taken when separating the focus styles between click and keyboard events as this may have unintended side effects for screen readers:

> Some browser/screen reader combinations fire mouse events, which could cause outlines to disappear while using this method.

https://a11yproject.com/posts/never-remove-css-outlines/

**Comment by taysea** [comment]
Agreed, I don't think we should separate the focusIndicator into multiple props for mouse vs keyboard. I think it would be best to look into why the focus indicator persists after clicking.

**Comment by iMerica** [comment]
> Disabling a focus indicator and or separating the styles between keyboard and mouse events inadvertently create less accessible components

Grommet is creating less accessible components because it leaves us with only one easy way to remove this on/after click outline: `focusIndicator={false}`.  This is having the unintended consequence of removing the outline for every type of event.

**Comment by iMerica** [comment]
When good intentions aren't grounded in reality:

`git grep focusIndicator={false} | wc -l`
> 36

**Comment by srallen** [comment]
x-posting here, I was incorrect about the focus persistence being a bug. This is what browsers do natively: https://github.com/grommet/grommet/issues/3656#issuecomment-574755913

**Comment by ShimiSun** [comment]
@steventilator I wouldn't recommend setting `focusIndicator={false}` as it will cause accessibility limitations. 

The reason the outline persists is that the interactive element still maintain the focus after clicking - and the focus will be placed on that DOM element until it will be directed elsewhere.
This standard is coming from WCAG 1.4.11, for reference see the following WebAIM examples https://webaim.org/temp/1-4-11examples.htm

For more references, check out the mozilla example @srallen had shared above.

Regarding the inconsistent browser behavior, It will help if you can provide a reproducible example, I've checked few of our examples and it seems that the outline persists on different browsers as expected.

**Comment by iMerica** [comment]
For anyone arriving here via Google, Slack, here is a temporary fix:

```css
button {
  outline: none !important;
  box-shadow: none !important;
}
```

**Comment by steventilator** [comment]
@ShimiSun I agree that setting `focusIndicator={false}` is not the smartest thing to do. Having accessibility baked into the framework is one of Grommet's great assets.

Nevertheless, I don't see how an outline that persists after the click is desired behavior. And so does our client. So, if she doesn't like it – and she doesn't, because she's not used to seeing it anywhere else on the web – in the end what's happening is that we disable the focusIndicator in its entirety. I'm sure we're not the only ones doing it this way.

As for the WCAG and MDN examples, I don't see an outline that persists after a mouse click on any of those buttons (using Firefox Developer Edition 73.0b9).

**Comment by srallen** [comment]
How the native focus outline appears varies with browser and OS. With MacOS and Chrome, you should see a blue focus outline persist with the native button behavior:

![focus-outline](https://user-images.githubusercontent.com/5016731/73290690-7a54e180-41c4-11ea-88c8-e90de9ecfac5.gif)

Firefox uses a dashed outline that I personally find harder to see. On further research, apparently the implementation of behavior is all over the place depending on OS and browser combination and may be contributing to the confusions and disagreements in this thread: https://zellwk.com/blog/inconsistent-button-behavior/

> Firefox on Mac is simply weird. Everything seems wrong.

Having the outline persist doesn't make sense for people who don't use accessibility tools, but it does for those who do. The outline itself is not required, but some kind of consistent styling with good contrast is, so if you do disable the focus outline, replace it with something else that still meets accessibility guidelines for the interaction and keep the styles consistent between mouse and keyboard since some users may alternate between modes of input. IMO, outline is the easiest to implement and to keep consistent between various interactions.

**Comment by hachok** [comment]
you can add a property to the global theme to style/remove focus from elements 

```
export const grommetTheme: ThemeType = {
  global: {
    focus: {
      border: undefined,
      outline: undefined,
      shadow: undefined
    },
...
}
```

**Comment by ShimiSun** [comment]
Grommet thrives to set its components for the correct accessibility behavior so the focus behavior will still maintain to be complaint with the WCAG standards.

For ways to make the border color transparent or change its color review the following story examples:
https://storybook.grommet.io/?path=/story/theme--focus
https://storybook.grommet.io/?path=/story/theme--themecontext-extend

**Comment by iMerica** [comment]
@ShimiSun 

> correct accessibility behavior

You've misunderstood the purpose of the WCAG. Correctness is something we can expect from mathematical proofs and type systems, not a qualitative spec like the WCAG with levels of advice (hint: the "G" in WCAG stands for "guidance"). 

Overall I get the sense that no one maintaining Grommet is applying independent critical thinking to these aesthetic bugs. Instead what I see is cargo culting a single interpretation of the WCAG in a rigid and fundamentalist way. 

Thanks,

Michael

**Comment by jjalonso** [comment]
that permanent focus outline is definitely the most hated feature xD
We should have a way to default opt out in theme.

**Comment by Ericky14** [comment]
This can be fixed by using this css style:

```
*:hover:not(:focus-visible),
*:focus:not(:focus-visible) {
  box-shadow: none;
  outline: none;
}
```

**Comment by jjalonso** [comment]
i have in my theme: inside globals

focus: {
border: {
color: 'transparent',
}, }
